### PR TITLE
Add timestamps to Groups and Templates

### PR DESCRIPTION
**Pull request by cheapRoc**

### CODE REVIEW

**Comment by cheapRoc** [comment]
This needs code review because I had to touch up quite a few conflicts while rebasing.

**Comment by kwilczynski** [comment]
Hi @cheapRoc! I am not sure about what could possibly go wrong after the merge conflicts were resolved, hence it needs another pair of eyes (Paul), but otherwise everything looks sound. :)

**Comment by cheapRoc** [comment]
@kwilczynski Mostly the queries that were changed but I think they look fine as well.

File: dev/backup.sql [hunk 1]
```diff
@@ -12,18 +12,18 @@ VALUES ('1d32f239-81e2-4e35-a258-a5649dc4e6f3', 'TSG_Management', '5a:ce:1e:1d:b
...
 
-INSERT INTO tsg_groups (id, "name", template_id, account_id, capacity, health_check_interval, archived) VALUES
-    ('9e075e5d-60d5-4cff-968e-b70db0badc12', 'test-group-1', 'ad74301e-ad62-404a-be44-3b2f24d082ac', '6f873d02-172c-418f-8416-4da2b50d5c53', 3, 300, false),
-    ('77135218-9e49-4ef7-81da-09de9ec580ff', 'test-group-2', 'f1ead2a9-92fc-4435-9eb8-9e520bc3e4f9', '6f873d02-172c-418f-8416-4da2b50d5c53', 3, 300, false),
-    ('95fb339f-8f8d-4184-ac5f-2c57e838136e', 'test-group-3', '93a4a267-498c-4911-a463-196eca9a5d99', '6f873d02-172c-418f-8416-4da2b50d5c53', 6, 60, false),
-    ('9dd64b6a-3d2c-4e92-b280-ef0f6cf49d04', 'test-group-4', 'deee2b55-11ef-4ffa-b34d-d1035ae1943b', '6f873d02-172c-418f-8416-4da2b50d5c53', 1, 300, false),
-    ('e6e5bc41-204f-4729-8189-ae3ec6385e95', 'test-group-5', '8b5a6001-8a59-4d85-bc72-1af83015b2c2', '6f873d02-172c-418f-8416-4da2b50d5c53', 3, 120, false),
-    ('1398d5b9-5750-4ed8-af0d-fe328f5d8cd0', 'test-group-6', '437c560d-b1a9-4dae-b3b3-6dbabb7d23a7', '6f873d02-172c-418f-8416-4da2b50d5c53', 12, 300, false);
+INSERT INTO tsg_groups (id, "name", template_id, account_id, capacity, health_check_interval, created_at, updated_at, archived) VALUES
```
Comment (kwilczynski) [inline dev/backup.sql]: I believe, the `health_check_interval` should be removed.

File: groups/groups_db.go [hunk 2]
```diff
@@ -88,21 +114,32 @@ func FindGroupByName(ctx context.Context, name string, accountID string) (*Servi
...
+	var (
+		group     ServiceGroup
+		createdAt pgtype.Timestamp
+		updatedAt pgtype.Timestamp
+	)
 
-	sqlStatement := `SELECT id, name, account_id, template_id, capacity
-FROM triton.tsg_groups
+	sqlStatement := `
+SELECT id, name, account_id, template_id, capacity, health_check_interval, created_at, updated_at
```
Comment (kwilczynski) [inline groups/groups_db.go]: I believe, the `health_check_interval` should be removed (we only keep it in the schema, etc.).

File: templates/template_db.go [hunk 3]
```diff
@@ -24,8 +24,8 @@ func FindTemplateByName(ctx context.Context, key string, accountID string) (*Ins
 		return nil, false
 	}
 
-	sqlStatement := `SELECT id, template_name, package, image_id, account_id, firewall_enabled, networks, COALESCE(metadata,''), userdata, COALESCE(tags,'')
-FROM triton.tsg_templates
+	sqlStatement := `SELECT id, template_name, package, image_id, account_id, firewall_enabled, networks, COALESCE(metadata,''), userdata, COALESCE(tags,''), created_at
```
Comment (kwilczynski) [inline templates/template_db.go]: Do we want to break on the back-tick (`) like the other statements? This is more for a sake of consistency.

File: dev/backup.sql (hunk 1 above)
Comment (cheapRoc) [reply dev/backup.sql]: I don’t believe I considered that while fixing the conflicts. I’ll make sure it’s still in the current scheme definition.

Comment (cheapRoc) [reply dev/backup.sql]: This does look correct to me.

**Comment by cheapRoc** [comment]
All the required changes are in order. Safe to merge for finalization of the client work.

**Comment by stack72** [comment]
This LGTM - I really liked that you used a `pgtype.Timestamp` and then merged it into the Group and Template later

**Pull request merged**